service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -58,7 +58,12 @@ public static boolean isValidOrigin(String origin, CorsSupportProperties corsSup
     // We want to strip the protocol.
     try {
       URL url = new URL(origin);
-      stringToCheck = url.getHost();
+      if (url.getPort() != -1) {
+        stringToCheck = url.getHost() + ':' + url.getPort();
+      } else {
+        stringToCheck = url.getHost();
+      }
+      logger.info(url.getPath());
     } catch (MalformedURLException e) {
       stringToCheck = origin;
```

Comment:
If we handled the wildcard case and the empty case before the try, we'd be able to simply fail out from here, right?